pyecm.py

```diff
@@ -1,1483 +1,1569 @@
+        alpha = 1
+        beta = 0
 
-      b_orig = b
-      alpha = 1
-      beta = 0
-
-      while not a & 1:
-         if alpha & 1:
-            alpha += b_orig
-         alpha >>= 1
-         a >>= 1
-
-      if b > a:
-         a, b = b, a
-         alpha, beta = beta, alpha
-
-      while b != 0 and a != b:
-         a -= b
-         alpha -= beta
-
-         while not a & 1:
+        while not a & 1:
             if alpha & 1:
-               alpha += b_orig
+                alpha += b_orig
             alpha >>= 1
             a >>= 1
 
-         if b > a:
-            a,b = b,a
+        if b > a:
+            a, b = b, a
             alpha, beta = beta, alpha
 
-      if a == b:
-         a -= b
-         alpha -= beta
-         a, b = b, a
-         alpha, beta = beta, alpha
-
-      if a != 1:
-         return 0
-
-      if truth:
-         alpha = b_orig - alpha
+        while b != 0 and a != b:
+            a -= b
+            alpha -= beta
 
-      return alpha
+            while not a & 1:
+                if alpha & 1:
+                    alpha += b_orig
+                alpha >>= 1
+                a >>= 1
 
-   def next_prime(n):
-      '''Finds the next prime after n.
+            if b > a:
+                a, b = b, a
+                alpha, beta = beta, alpha
 
-Returns the next prime after n.'''
-      n += 1
-      if n <= 167:
-         if n <= 23:
-            if n <= 3:
-               return 3 - (n <= 2)
-            n += (n & 1) ^ 1
-            return n + (((4 - (n % 3)) >> 1) & 2)
-
-         n += (n & 1) ^ 1
-         inc = n % 3
-         n += ((4 - inc) >> 1) & 2
-         inc = 6 - ((inc + ((2 - inc) & 2)) << 1)
+        if a == b:
+            a -= b
+            alpha -= beta
+            a, b = b, a
+            alpha, beta = beta, alpha
 
-         while 0 in (n % 5, n % 7, n % 11):
-            n += inc
-            inc = 6 - inc
-         return n
+        if a != 1:
+            return 0
 
-      n += (n & 1) ^ 1
-      inc = n % 3
-      n += ((4 - inc) >> 1) & 2
-      inc = 6 - ((inc + ((2 - inc) & 2)) << 1)
-      should_break = False
+        if truth:
+            alpha = b_orig - alpha
 
-      while 1:
-         for prime in PRIMES:
-            if not n % prime:
-               should_break = True
-               break
+        return alpha
 
-         if should_break:
-            should_break = False
-            n += inc
-            inc = 6 - inc
-            continue
+    def next_prime(n):
+        """
+        Finds the next prime after n.
+        Returns the next prime after n.
+        """
+        n += 1
+        if n <= 167:
+            if n <= 23:
+                if n <= 3:
+                    return 3 - (n <= 2)
+                n += (n & 1) ^ 1
+                return n + (((4 - (n % 3)) >> 1) & 2)
 
-         p = 1
-         for i in range(int(math.log(n) / LOG_2), 0, -1):
-            p <<= (n >> i) & 1
-            p = (p * p) % n
+            n += (n & 1) ^ 1
```

Comment:
Why did you remove the for loop here?

Reply:
It's still there, in 180. line.

Reply:
OK; github's diffs make it hard to tell!